src/components/Global/ButtonBasic.vue

```diff
@@ -12,14 +12,17 @@ export default defineComponent({
       type: String,
       required: true,
     },
+    class: {
+      type: String,
+    },
     colorTheme: {
       type: String,
       default: "red-white",
```

Comment:
I know this is not the part of this PR (and it's still in progress) buuuut.
If you will be dealing with a list of known values (for example the different variants of a button) we could throw in a validator here to restrict us only to such list:
https://vuejs.org/guide/components/props.html#prop-validation
This way if someone would pass an improper value - they would get an error 😄